ENIGMAsystem/SHELL/Universal_System/Serialization/detect_serialization.h

```diff
@@ -0,0 +1,187 @@
+/** Copyright (C) 2022 Dhruv Chawla
+*** Copyright (C) 2023 Fares Atef
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+/**
+  @file detect_serialization.h
+  @brief This file contains macros that are used to detect whether a class has a specific method or not
+  (INTERNAL_HAS_MEMBER_FUNCTION), whether a class has a specific static function or not(HAS_STATIC_FUNCTION),
+  and whether a class has a specific free function or not(HAS_FREE_FUNCTION).
+*/
+
+#ifndef ENIGMA_DETECT_SERIALIZATION_H
+#define ENIGMA_DETECT_SERIALIZATION_H
+
+// It is defined in var4.h, and has conflicts with the std::string.
```

Comment:
It is defined [here](https://github.com/FaresAtef1/enigma-dev/blob/GSoC-Part3/ENIGMAsystem/SHELL/Universal_System/var4.h#L266).